lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,17 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          return false if ends_with_equals? sym_name
+          valid_19_syntax_symbol? sym_name
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
+          sym_name.sub!(/\A:/, '')
+          RuboCop::ProcessedSource.new("{ #{sym_name}: :foo }").valid_syntax?
+        end
+
+        def ends_with_equals?(sym_name)
```

Comment:
This method is wrongly named – it does more that just check the ending of the string.

Anyhow, I think that with the clever `valid_syntax` approach above, the checks in this method aren’t even needed anymore.